src/engine/shared/cfgvar_buffer.cpp

```diff
@@ -72,73 +162,98 @@ void CCfgVarBuffer::Init()
+		"cfg var changes between this start command and the stop command will be reset at the end of next round");
+	pConsole->Register("resetcfg_nextround_end", "", CFGFLAG_SERVER, ConResetCfgNextRound_End, pConsole, 
+		"cfg var changes between the start command and this stop command will be reset at the end of next round");
+	pConsole->Register("print_round_cfg", "?s", CFGFLAG_SERVER, ConPrintRoundCfg, pConsole, 
+		"show round specifc config vars, format: config_var original_value -> temp_value");
+}
 
-void CCfgVarBuffer::ConPrintCfg(CConsole* pConsole, const char *pCfgName)
+bool CCfgVarBuffer::IsConfigVar(const char* pStr)
 {
-	if (*pCfgName == 0) 
+	for (int i = 0; i < m_CfgVarsNum; i++)
 	{
-		// print all config variables and their values
-		for (int i = 0; i < m_CfgVarsCounter; i++)
-		{
-			char lineBuff[512];
-			if (m_CfgVars[i].m_Type == CFG_TYPE_INT)
-				str_format(lineBuff, 512, "%s %i", m_CfgVars[i].m_pScriptName, *m_CfgVars[i].m_pIntValue);
-			else
-				str_format(lineBuff, 512, "%s %s", m_CfgVars[i].m_pScriptName, m_CfgVars[i].m_pStrValue);
-			pConsole->Print(IConsole::OUTPUT_LEVEL_STANDARD, "Console", lineBuff);
-		}
-		return;
+		if (str_comp_nocase(m_pCfgVars[i].m_pScriptName, pStr) == 0) return true;
 	}
+	return false;
+}
 
-	// search for config vars that contain pCfgName and print them and their values
-	for (int i = 0; i < m_CfgVarsCounter; i++)
+void CCfgVarBuffer::ConPrintCfg(CConsole* pConsole, const char *pCfgName)
+{
+	// search for config vars that contain pCfgName and print them and their values - if pCfgName is NULL print all vars
+	for (int i = 0; i < m_CfgVarsNum; i++)
 	{
-		bool contains = false;
-		int m = 0;
-		for (int k = 0; k < m_CfgVars[i].m_ScriptNameLength; k++)
-		{
-			if (pCfgName[m] == m_CfgVars[i].m_pScriptName[k])
-				m++;
-			else 
-				m = 0;
-			if (pCfgName[m] == 0)
-			{
-				contains = true;
-				break;
-			}
-		}
-		if (!contains) continue;
+		if (pCfgName && pCfgName[0] != 0)
+			if (!strstr(m_pCfgVars[i].m_pScriptName, pCfgName)) continue;
 
 		char lineBuff[512];
-		if (m_CfgVars[i].m_Type == CFG_TYPE_INT)
-			str_format(lineBuff, 512, "%s %i", m_CfgVars[i].m_pScriptName, *m_CfgVars[i].m_pIntValue);
+		if (m_pCfgVars[i].m_Type == CFG_TYPE_INT)
+			str_format(lineBuff, 512, "%s %i", m_pCfgVars[i].m_pScriptName, *m_pCfgVars[i].m_pIntValue);
 		else
-			str_format(lineBuff, 512, "%s %s", m_CfgVars[i].m_pScriptName, m_CfgVars[i].m_pStrValue);
+			str_format(lineBuff, 512, "%s %s", m_pCfgVars[i].m_pScriptName, m_pCfgVars[i].m_pStrValue);
 		pConsole->Print(IConsole::OUTPUT_LEVEL_STANDARD, "Console", lineBuff);
 	}
 }
 
-bool CCfgVarBuffer::IsConfigVar(const char* pStr)
+bool CCfgVarBuffer::ConResetCfgNextRound_Start(IConsole::IResult *pResult, void *pUserData)
+{
+	//((CConsole*)pUserData)->Print(IConsole::OUTPUT_LEVEL_STANDARD, "Console", "Reset upcomming config variables at the end of next round");
```

Comment:
It's for debug purposes. I also uncomment these lines, because they can be useful for debugging. 
We could introduce a verbose flag set during the compile-process, which would activate this line.